plugins/libvirt/libvirt.go

```diff
@@ -0,0 +1,71 @@
+package libvirt
+
+import (
+	lv "github.com/alexzorin/libvirt-go"
+	"github.com/influxdb/telegraf/plugins"
+)
+
+const sampleConfig = `
+	# specify a libvirt connection uri
+	uri = "qemu:///system"
+`
+
+type Libvirt struct {
+	Uri string
+}
+
+func (l *Libvirt) SampleConfig() string {
+	return sampleConfig
+}
+
+func (l *Libvirt) Description() string {
+	return "Read domain infos from a libvirt deamon"
+}
+
+func (l *Libvirt) Gather(acc plugins.Accumulator) error {
+	connection, err := lv.NewVirConnectionReadOnly(l.Uri)
+	if err != nil {
+		return err
+	}
+	defer connection.CloseConnection()
+
+	nodeName, _ := connection.GetHostname()
+
+	domains, err := connection.ListDomains()
+	if err != nil {
+		return err
+	}
+
+	for _, domainId := range domains {
+		domain, err := connection.LookupDomainById(domainId)
+		if err != nil {
+			return err
+		}
+
+		domainName, _ := domain.GetName()
+		tags := map[string]string{"host": domainName, "node": nodeName}
```

Comment:
Not intentional, no. I'll rename it to "domain" which follows more the libvirt terminology.